Hace falta algo que indique con qué entorno vamos a trabajar

Importar lo que hace falta

In [1]:
import pandas as pd

In [2]:
data_url = "../Data/properatti.csv"

In [3]:
data = pd.read_csv(data_url, encoding="utf-8")

In [4]:
data.shape

(121220, 26)

In [5]:
data.isnull().sum()

Unnamed: 0                         0
operation                          0
property_type                      0
place_name                        23
place_with_parent_names            0
country_name                       0
state_name                         0
geonames_id                    18717
lat-lon                        51550
lat                            51550
lon                            51550
price                          20410
currency                       20411
price_aprox_local_currency     20410
price_aprox_usd                20410
surface_total_in_m2            39328
surface_covered_in_m2          19907
price_usd_per_m2               52603
price_per_m2                   33562
floor                         113321
rooms                          73830
expenses                      106958
properati_url                      0
description                        2
title                              0
image_thumbnail                 3112
dtype: int64

Confirmar los valores de la serie 'currency'

La serie price_aprox_usd tiene 20410 null, se eliminan esas observaciones 

In [6]:
data = data[data['price_aprox_usd'].notnull()]

In [7]:
data.shape

(100810, 26)

In [8]:
data.isnull().sum()

Unnamed: 0                        0
operation                         0
property_type                     0
place_name                       23
place_with_parent_names           0
country_name                      0
state_name                        0
geonames_id                   16952
lat-lon                       42262
lat                           42262
lon                           42262
price                             0
currency                          1
price_aprox_local_currency        0
price_aprox_usd                   0
surface_total_in_m2           31944
surface_covered_in_m2         13150
price_usd_per_m2              32193
price_per_m2                  13152
floor                         93894
rooms                         61231
expenses                      87564
properati_url                     0
description                       2
title                             0
image_thumbnail                2250
dtype: int64

**El objetivo son los valores en la serie 'price_aprox_usd'**

In [11]:
data = data.drop(['currency', 'price', 'price_aprox_local_currency'], axis='columns')

In [12]:
data.shape

(100810, 23)

In [13]:
data.isnull().sum()

Unnamed: 0                     0
operation                      0
property_type                  0
place_name                    23
place_with_parent_names        0
country_name                   0
state_name                     0
geonames_id                16952
lat-lon                    42262
lat                        42262
lon                        42262
price_aprox_usd                0
surface_total_in_m2        31944
surface_covered_in_m2      13150
price_usd_per_m2           32193
price_per_m2               13152
floor                      93894
rooms                      61231
expenses                   87564
properati_url                  0
description                    2
title                          0
image_thumbnail             2250
dtype: int64